### PR TITLE
Improved 3d Layout methods to get location

### PR DESCRIPTION
**Pull request by maxcapodi78**
Added 3d Layout component bounding_box property
Added 3d layout component clearance creation on soldermask

### CODE REVIEW

File: pyaedt/application/Design.py [hunk 1]
```diff
@@ -156,6 +158,17 @@ def __init__(
         port=0,
         aedt_process_id=None,
     ):
+        def load_aedt_thread(path):
+            start = time.time()
+            settings._project_properties[path] = load_entire_aedt_file(path)
+            settings._project_time_stamp = os.path.getmtime(project_name)
+            logger = logging.getLogger("Global")
+            logger.info("AEDT thread file load time {}".format(time.time() - start))
```
Comment (Alberto-DM) [inline pyaedt/application/Design.py:166]:
```suggestion
            logger.info("AEDT file load (threaded) time: {}".format(time.time() - start))
```

File: pyaedt/modeler/object3dlayout.py [hunk 2]
```diff
@@ -340,13 +485,19 @@ def enabled(self, status=True):
...
+        comp_info = self.m_Editor.GetComponentInfo(self.name)
+        for el in comp_info:
+            if "ComponentProp=" in el and "CompPropEnabled=false" in el:
+                return False
+            elif "ComponentProp=" in el and "CompPropEnabled=true" in el:
+                return True
+        return True
+
+    @enabled.setter
+    def enabled(self, status=True):
```
Comment (Alberto-DM) [inline pyaedt/modeler/object3dlayout.py:497]:
```suggestion
    def enabled(self, status):
```
I think you should not put a default argument value in the setter. Maybe does't give an error, but it's never used.

**Review by Alberto-DM — APPROVED** [review]
Just 2 comments, but otherwise LGTM.

File: pyaedt/application/Design.py (hunk 1 above)
Comment (maxcapodi78) [reply pyaedt/application/Design.py]: Fixed. Thanks

File: pyaedt/modeler/object3dlayout.py (hunk 2 above)
Comment (maxcapodi78) [reply pyaedt/modeler/object3dlayout.py]: Done

**Comment by PipKat** [comment]
Maxime resolved the Vale error yesterday but making the heading in the error sentence case. Updating this branch should resolve it.

**Pull request merged**